handlers/grafana.go

```diff
@@ -73,14 +73,20 @@ func getGrafanaInfo(requestToken string, dashboardSupplier dashboardSupplier) (*
 
 	// Call Grafana REST API to get dashboard urls
 	links := []kmodel.ExternalLink{}
+	urlParts := strings.Split(externalURL, "?")
```

Comment:
Is this expected format documented in Kiali CR template ?
(It should be the main source for these formats doc, also good if kiali.io can be updated).

Reply:
What do you mean? I don't think there's anything to change in CR documentation, simple URL will continue to work as before. It's just that now we *also* accept URLs that contain query params.

Reply:
Yes, specially as "http://grafana:3000?orgId=1" is a non standard format and it should be well documented to redirect users when something goes wrong.

Reply:
This needs to be updated:
https://github.com/kiali/kiali-operator/blob/master/deploy/kiali/kiali_cr.yaml#L405

How the user would know that this format is available ?

Reply:
but the point is that it's not wrong anymore with this PR. Would you like us to say in doc: "if you set an URL with query params, such as http://grafana:3000?orgId=1 , it's going to work as well" ?

Reply:
I think it would be good to mention that in the Kiali CR comment.

Reply:
Also I don't understand why you say it's non standard. Query params in URL are standard (?)

Reply:
> And what is the expected one ? "http://grafana:3000" ? "http://grafana:3000/" ? "http://grafana:3000/?orgId=1" ?

All are correct, that's my point. With this PR we're not tightening expectations, on the contrary, we're relaxing them.

Reply:
Note that when we write `urlParts := strings.Split(externalURL, "?")` , if the URL doesn't contain `?`, then it all goes in `urlParts[0]`, that's why it doesn't break the existing format.

Reply:
There's also a couple of tests in `grafana_test.go` to ensure old formats continue to work.

Reply:
I don't understand why user need to guess that static parameters are allowed.

What I'm saying is that some similar comment in the Kiali CR (because the CR is the source of truth for config documentation) like

"Users may want to add static query parameters such as org id:
http://mygrafana:3000?orgId=1"

Reply:
Ok, I'm going to add that, although I'm not really convinced. Query params *are* standard in URL, so to me the fact that we expect an URL should implicitly mean it's allowed.

This is the opposite that would be problematic IMO: if setting query params in a URL would break anything, that should be mentioned because it means we don't accept any kind of URL. Which is the current situation for Grafana without this PR, and perhaps (would need to check) the situation with jaeger as well. I believe it's more important to tell when we don't follow standard, than when we do follow it.

Anyway, it's not a big deal to add it to doc, so i'll do it.

Reply:
hmm looks like it's in the operator repo so it's going to be a separate PR anyway

Reply:
BTW if we start documenting to that level of granularity, we should do the same for everywhere we expect an URL: tell if query params are allowed or not?

Reply:
If someone has a reasonable doubt about a config parameter, asking to add some comment in the Kiali CR is always good.

We have not started anything new. 
We are reviewing a PR and there was a comment about the format of a URL when we have several comments about what is the well formed one.

Or perhaps I don't understand your question or the intention.